ruby/ql/lib/codeql/ruby/frameworks/ActiveStorage.qll

```diff
@@ -8,23 +8,236 @@ private import codeql.ruby.Concepts
 private import codeql.ruby.DataFlow
 private import codeql.ruby.dataflow.FlowSummary
 private import codeql.ruby.frameworks.data.ModelsAsData
+private import codeql.ruby.frameworks.ActiveRecord
 
-/** A call to `ActiveStorage::Filename#sanitized`, considered as a path sanitizer. */
-class ActiveStorageFilenameSanitizedCall extends Path::PathSanitization::Range, DataFlow::CallNode {
-  ActiveStorageFilenameSanitizedCall() {
-    this.getReceiver() =
-      API::getTopLevelMember("ActiveStorage").getMember("Filename").getAnInstantiation() and
-    this.getMethodName() = "sanitized"
+/**
+ * Provides modeling for the `ActiveStorage` library.
+ * Version: 7.0.
+ */
+module ActiveStorage {
+  /** A call to `ActiveStorage::Filename#sanitized`, considered as a path sanitizer. */
+  private class FilenameSanitizedCall extends Path::PathSanitization::Range, DataFlow::CallNode {
+    FilenameSanitizedCall() {
+      this =
+        API::getTopLevelMember("ActiveStorage")
+            .getMember("Filename")
+            .getInstance()
+            .getAMethodCall("sanitized")
+    }
   }
-}
 
-/** Taint related to `ActiveStorage::Filename`. */
-private class Summaries extends ModelInput::SummaryModelCsv {
-  override predicate row(string row) {
-    row =
-      [
-        "activestorage;;Member[ActiveStorage].Member[Filename].Method[new];Argument[0];ReturnValue;taint",
-        "activestorage;;Member[ActiveStorage].Member[Filename].Instance.Method[sanitized];Argument[self];ReturnValue;taint",
-      ]
+  /** Taint related to `ActiveStorage::Filename`. */
+  private class FilenameSummaries extends ModelInput::SummaryModelCsv {
+    override predicate row(string row) {
+      row =
+        [
+          "activestorage;;Member[ActiveStorage].Member[Filename].Method[new];Argument[0];ReturnValue;taint",
+          "activestorage;;Member[ActiveStorage].Member[Filename].Instance.Method[sanitized];Argument[self];ReturnValue;taint",
+        ]
+    }
+  }
+
+  /**
+   * `Blob` is an instance of `ActiveStorage::Blob`.
+   */
+  private class BlobTypeSummary extends ModelInput::TypeModelCsv {
+    override predicate row(string row) {
+      // package1;type1;package2;type2;path
+      row =
+        [
+          // ActiveStorage::Blob.new : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Instance",
+          // ActiveStorage::Blob.create_and_upload! : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[create_and_upload!].ReturnValue",
+          // ActiveStorage::Blob.create_before_direct_upload! : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[create_before_direct_upload!].ReturnValue",
+          // ActiveStorage::Blob.compose(blobs : [Blob]) : Blob
+          "activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[compose].ReturnValue",
+          // gives error: Invalid name 'Element' in access path
```

Comment:
Does this work with an unparameterized `Element` (i.e. `"activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[compose].Argument[0].Element"`) or with `ArrayElement` instead (i.e. `"activestorage;Blob;activestorage;;Member[ActiveStorage].Member[Blob].Method[compose].Argument[0].ArrayElement"`)?